debug/org.eclipse.debug.core/core/org/eclipse/debug/core/model/RuntimeProcess.java

```diff
@@ -152,9 +152,9 @@ public RuntimeProcess(ILaunch launch, Process process, String name, Map<String,
 		fThreadNameSuffix = getPidInfo(process, launch);
 
 		fStreamsProxy = createStreamsProxy();
+		launch.addProcess(this);
```

Comment:
Please add a comment about the position being important here, or someone may easily miss it later and refactor in a regressing way.

Reply:
Thanks, @mickaelistria! Makes sense to document that. I have added a comment in e289a48.
In case someone misses the comment, there will still be the added regression test that will fail.

Reply:
It is not nice that the object exposes itself in constructor, but I guess this is nothing we will fix now.
May I suggest to **first** create a monitor (without starting it), just to finish init of the process fields that are supposed to be non-null, before adding the instance to the launch?

Reply:
I fully agree. From my point of view, the constructor does more than a constructor is supposed to do. The process creation might be done by a factory that manages proper instantiation by (1) creating a `RuntimeProcess` with proper information from the `Launch`, (2) registering it at the `Launch`, (3) starting the process (in terms of creating the monitor thread). Then the `RuntimeProcess` would not even have to know about a `Launch`. Actually `RuntimeProcess` is already instantiated in factory methods, but since it's public API, everyone can instantiate it at any place.

Changing the instruction order to finish intialization before adding the instance to the launch totally makes sense. I have adapted that in 1fed561.